### PR TITLE
Port to Vaadin 8

### LINKED ISSUE #2: Upgrade to Vaadin 8
Upgrade the add-on so that it would be possible to use in Vaadin 8 applications.

### PR DESCRIPTION
**Pull request by wbadam**
In the spirit of the March of add-ons I ported yours to Vaadin 8. Would be great if you could review and make a release to the Directory.

Resolves #2

Thanks,
Adam

### CODE REVIEW

File: KeyActions/src/main/java/org/vaadin/artur/client/KeyActionConnector.java [hunk 1]
```diff
@@ -44,12 +42,6 @@ protected void triggerEvent(final KeyDownEvent event) {
 		Scheduler.get().scheduleDeferred(new ScheduledCommand() {
 			@Override
 			public void execute() {
-				if (getParent() instanceof BeforeShortcutActionListener) {
```
Comment (Artur-) [inline KeyActions/src/main/java/org/vaadin/artur/client/KeyActionConnector.java:47]: I have a feeling that this needs to do `getParent().flush()` to be sure that the contents of e.g. a TextField is sent to the server before the key event

File: KeyActions/pom.xml [hunk 2]
```diff
@@ -6,11 +6,11 @@
 	<artifactId>KeyActions</artifactId>
 	<packaging>jar</packaging>
 	<name>KeyActions</name>
-        <version>0.2-SNAPSHOT</version>
+        <version>1.2-SNAPSHOT</version>
```
Comment (Artur-) [inline KeyActions/pom.xml]: I think 1.0-SNAPSHOT would make more sense

**Review by Artur- — CHANGES_REQUESTED** [review]
Thanks! Looks good

File: KeyActions/pom.xml (hunk 2 above)
Comment (wbadam) [reply KeyActions/pom.xml]: Fair enough

File: KeyActions/src/main/java/org/vaadin/artur/client/KeyActionConnector.java (hunk 1 above)
Comment (wbadam) [reply KeyActions/src/main/java/org/vaadin/artur/client/KeyActionConnector.java:47]: Do you mean call flush() before the RPC call? Please see the modified file if that's what you meant.

**Comment by Artur-** [comment]
Awesome

**Pull request merged**